File: Ahorn/Triggers/timeModulationTrigger.jl
module ShroomHelperTimeModulationTrigger

using ..Ahorn, Maple

@mapdef Trigger "ShroomHelper/TimeModulationTrigger" TimeModulationTrigger(
	x::Integer, 
	y::Integer, 
	width::Integer=Maple.defaultTriggerWidth, 
	height::Integer=Maple.defaultTriggerHeight, 
	timeFrom::Number=1.0, 
	timeTo::Number=1.0, 
	positionMode::String="LeftToRight",
	destroyOnLeave::Bool=false,
	persistent::Bool=false
)

const placements = Ahorn.PlacementDict(
    "Time Modulation Trigger (Shroom Helper)" => Ahorn.EntityPlacement(
        TimeModulationTrigger,
        "rectangle"
    )
)

Ahorn.editingOptions(entity::TimeModulationTrigger) = Dict{String, Any}(
    "positionMode" => Maple.trigger_position_modes
)

end

File: Loenn/triggers/time_modulation_trigger.lua
local enums = require("consts.celeste_enums")

local time_modulation_trigger = {}

time_modulation_trigger.name = "ShroomHelper/TimeModulationTrigger"

time_modulation_trigger.fieldInformation = {
    positionMode = {
        options = enums.trigger_position_modes,
        editable = false
    }
}

time_modulation_trigger.fieldOrder = { 
    "x", "y", "width", "height", "timeFrom", "timeTo", "positionMode", "destroyOnLeave", "persistent" 
}

time_modulation_trigger.placements = {
    name = "time_modulation_trigger",
    data = {
        timeFrom = 1.0,
        timeTo = 1.0,
        positionMode = "LeftToRight",
        destroyOnLeave = false,
        persistent = false,
    }
}

return time_modulation_trigger